server/mailet/mailets/src/test/java/org/apache/james/transport/mailets/ToSenderDomainRepositoryTest.java

```diff
@@ -0,0 +1,195 @@
+            .name("mailName")
+            .sender(MailAddressFixture.SENDER)
+            .state(Mail.DEFAULT)
+            .build();
+
+        mailet.service(mail);
+
+        assertThat(mail.getState())
+            .isEqualTo(Mail.GHOST);
+    }
+
+    @Test
+    void serviceShouldPreserveMailStateWhenPassThrough() throws Exception {
+        FakeMailetConfig mailetConfig = FakeMailetConfig.builder()
+            .mailetName("TestConfig")
+            .setProperty("passThrough", "true")
+            .setProperty("urlPrefix", "memory://var/mail/dlp/")
+            .build();
+        mailet.init(mailetConfig);
+
+        FakeMail mail = FakeMail.builder()
+            .name("mailName")
+            .sender(MailAddressFixture.SENDER)
+            .state(Mail.DEFAULT)
+            .build();
+
+        mailet.service(mail);
+
+        assertThat(mail.getState())
+            .isEqualTo(Mail.DEFAULT);
+    }
+
+    @Test
+    void passThroughShouldDefaultToFalse() throws Exception {
+        mailet.init(DEFAULT_MAILET_CONFIG);
+
+        FakeMail mail = FakeMail.builder()
+            .name("mailName")
+            .sender(MailAddressFixture.SENDER)
+            .state(Mail.DEFAULT)
+            .build();
+
+        mailet.service(mail);
+
+        assertThat(mail.getState())
+            .isEqualTo(Mail.GHOST);
+    }
+
+    @Test
+    void initShouldSetNotPassThroughWhenPassThroughtIsNotBooleans() throws Exception {
+        MailRepositoryStore mailRepositoryStore = mock(MailRepositoryStore.class);
+        ToSenderDomainRepository mailet = new ToSenderDomainRepository(mailRepositoryStore);
+        when(mailRepositoryStore.select(any()))
+            .thenThrow(new MailRepositoryStore.MailRepositoryStoreException("any"));
+
+        mailet.init(DEFAULT_MAILET_CONFIG);
+
+        FakeMail mail = FakeMail.builder()
+            .name("mailName")
+            .sender(MailAddressFixture.SENDER)
+            .state(Mail.DEFAULT)
+            .build();
+
+        assertThatThrownBy(() -> mailet.service(mail))
+            .isInstanceOf(MessagingException.class);
+    }
+
+    @Test
+    void initShouldSetNotPassThroughWhenPassThroughtIsNotBoolean() throws Exception {
+        FakeMailetConfig mailetConfig = FakeMailetConfig.builder()
+            .mailetName("TestConfig")
+            .setProperty("urlPrefix", "memory://var/mail/dlp/")
+            .setProperty("passThrough", "not boolean")
+            .build();
+
+        mailet.init(mailetConfig);
+
+        FakeMail mail = FakeMail.builder()
+            .name("mailName")
+            .sender(MailAddressFixture.SENDER)
+            .state(Mail.DEFAULT)
+            .build();
+
+        mailet.service(mail);
+
+        assertThat(mail.getState())
+            .isEqualTo(Mail.GHOST);
+    }
+
+    @Test
+    void getMailetInfoShouldReturnExpectedResult() {
+        assertThat(mailet.getMailetInfo())
+            .isEqualTo("ToSenderDomainRepository Mailet");
+    }
```

Comment:
can you at least add a test when the sender domain repository does not exist?

Reply:
Not possible with `MockMailRepositoryStore` very sadly... You do have to provide the repository.

Mail repository store is bould to the injection framework which proposes two different implementation. So in unit tests we can not demonstrate this.

The solution might be to write a quick integration test. We will work on it tomorrow with Duc.

Reply:
The was a fault, the method name `initShouldSetNotPassThroughWhenPassThroughtIsNotBoolean` is duplicated, the test that @rouazana want may be described the method `initShouldSetNotPassThroughWhenPassThroughtIsNotBooleans`